receiver/windowsperfcountersreceiver/windowsperfcounters_scraper.go

```diff
@@ -157,9 +157,21 @@ func (s *scraper) scrape(context.Context) (pmetric.Metrics, error) {
 			initializeMetricDps(metric, now, val, watcher.MetricRep.Attributes)
 		}
 	}
+
+	// Drop metrics with no datapoints. This happens when configured counters don't exist on the host.
+	// This may result in a Metrics message with no metrics if all counters are missing.
+	metricSlice.RemoveIf(func(m pmetric.Metric) bool {
+		if m.Type() == pmetric.MetricTypeGauge {
+			return m.Gauge().DataPoints().Len() == 0
+		} else {
+			return m.Sum().DataPoints().Len() == 0
+		}
```

Comment:
Let's use switch/case and check for the Sum type explicitly. Even if we don't collect anything else right now, we want to avoid potential panics if something else is added in future

Reply:
I've made it so that unknown types don't remove empty datapoints for other metric types. 

I think there are a few other spots where the receiver assumes that everything in perfmon can only be a sum or a gauge. It's fairly unlikely that Windows Perfmon ever adds histograms, but on the off chance that it does happen, I think the receiver will need significant refactoring to support them.